### PR TITLE
Update Spark Iceberg image to v0.13.1

### PR DESCRIPTION
**Pull request by alexjo2144**
Required for testing v2 tables with merge-on-read deletes.

### CODE REVIEW

File: testing/spark3.0-iceberg/Dockerfile [hunk 1]
```diff
@@ -12,9 +12,10 @@
 
 FROM testing/centos7-oj11:unlabelled
 
-ARG SPARK_VERSION=3.1.2
+ARG SPARK_VERSION=3.2.1
 ARG HADOOP_VERSION=3.2
-ARG ICEBERG_VERSION=0.12.1
+ARG ICEBERG_VERSION=0.13.1
+ARG ICEBERG_JAR_VERSION=3.2_2.12
```
Comment (findepi) [inline testing/spark3.0-iceberg/Dockerfile:18]: spark 3.2.1 vs iceberg-spark 3.2.2 -- should these match?

Comment (alexjo2144) [reply testing/spark3.0-iceberg/Dockerfile:18]: 2.12 is the Scala version number, not the spark minor version. They're both referring to Spark 3.2.1

Comment (findepi) [reply testing/spark3.0-iceberg/Dockerfile:18]: worth adding a code comment?

Comment (alexjo2144) [reply testing/spark3.0-iceberg/Dockerfile:19]: Added a comment

**Comment by findepi** [comment]
release as 55

**Comment by findepi** [comment]
@alexjo2144 can you please also rename the image to `spark3-...` ?
it looks like an omission in https://github.com/trinodb/docker-images/pull/119, but I would be grateful if you could fix that

File: testing/spark3.0-iceberg/Dockerfile (hunk 1 above)
Comment (rashworld-max) [reply testing/spark3.0-iceberg/Dockerfile:19]: T

File: testing/spark3.0-iceberg/Dockerfile [hunk 2]
```diff
@@ -12,9 +12,11 @@
 
 FROM testing/centos7-oj11:unlabelled
 
-ARG SPARK_VERSION=3.1.2
+ARG SPARK_VERSION=3.2.1
 ARG HADOOP_VERSION=3.2
-ARG ICEBERG_VERSION=0.12.1
+ARG ICEBERG_VERSION=0.13.1
+# ICEBERG_JAR_VERSION is derived from: <spark-version>_<scala-version>
+ARG ICEBERG_JAR_VERSION=3.2_2.12
```
Comment (rashworld-max) [inline testing/spark3.0-iceberg/Dockerfile:19]: T

**Review by rashworld-max — APPROVED** [review]

**Pull request merged**